### PR TITLE
Add iterator to skip first value if equals to after

### LINKED ISSUE #3513: Utillity for `after` and `start` params for the store Scan
The store API consist of a `Scan(ctx context.Context, partitionKey, start []byte) (EntriesIterator, error)`, while our api and internal wiring uses the `after` notation. We need to have a easy way to convert the returned EntriesIterator to skip that `start` key, if it exists.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #3513

### CODE REVIEW

**Review by N-o-Z — CHANGES_REQUESTED** [review]
I wonder what use case would we have for PrefixIterator without the skip logic. I can't seem to think of any. If it makes sense, I think this logic should be in the PrefixIterator - don't see a reason to wrap in a new iterator

File: pkg/kv/skip_iterator.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package kv
+
+import "bytes"
+
+type SkipIterator struct {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/kv/skip_iterator.go:5]: It is easier to have a single functionality for every iterator, and then compose them in a way that fits the use-case.
For example, we might need a `PrefixIterator` with a `start` and not an `after` notation.

**Comment by itaiad200** [comment]
> I wonder what use case would we have for PrefixIterator without the skip logic. I can't seem to think of any. If it makes sense, I think this logic should be in the PrefixIterator - don't see a reason to wrap in a new iterator

What's the downside of a new iterator? With single responsibility per iterator the code remains more readable. Even with the fairly simple logics we have now, prefix & skipping the first, the `Next()` call becomes complex.

**Review by eden-ohana — APPROVED** [review]
Thanks! can we document the new iterator behavior with the after key

**Review by N-o-Z — APPROVED** [review]

**Pull request merged**